File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const stack = [];
    
    for (const item of s) {
        // stack이 비어있으면 추가
        if (!stack.length) {
            stack.push(item);
        } else if (stack[stack.length - 1] === "(" && item === ')') {
            stack.pop();
        } else {
            stack.push(item);
        }
    }
    
    if (stack.length) {
        return false;
    }
    return true;
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    stack = []
    
    for i in s:
        if not len(stack):
            stack.append(i)
        else:
            if i == ')' and stack[-1] == "(":
                stack.pop()
            else:
                stack.append(i)
                
    return True if not len(stack) else False